projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx

```diff
@@ -332,6 +332,12 @@ export function VideoHoverPreviewControl( {
 		Math.min( MAX_LOOP_DURATION, videoDuration - previewAtTime )
 	);
 
+	useEffect( () => {
+		if ( loopDuration > maxLoopDuration ) {
+			onLoopDurationChange( maxLoopDuration );
+		}
+	}, [ maxLoopDuration ] );
```

Comment:
The callback function should handle it, IMO. That's why I suggest that it may have only one callback to take both data: starting point and duration. It could be doable by passing an object.

Reply:
Instead of using the effect hook, I think it should be handled when the starting point changes, it's in the callback passed to the onDebounceChange property. Something like:

```es6
<TimestampControl
	label={ __( 'Starting point', 'jetpack-videopress-pkg' ) }
	onDebounceChange={ timestamp => {
		onPreviewAtTimeChange( timestamp );
		const max = Math.min( MAX_LOOP_DURATION, videoDuration - timestamp );

		setMaxLoopDuration( max );

		// Adjust loop duration if it's needed
		if ( loopDuration > max ) {
			onLoopDurationChange( max );
		}
	} }
/>
```

Reply:
> The callback is only called when the value changes, not when the max value changes,

The `max` value is updated when the _value_(the starting point `previewAtTime`) changes:

```es6
const max = Math.min( MAX_LOOP_DURATION, videoDuration - timestamp );
setMaxLoopDuration( max );
```

>  there is no callback for that.

No, there isn't. The component has two properties for that: `onPreviewAtTimeChange` and `onLoopDurationChange`

Considering that the max value of the loop duration can change, we should recalculate the loop duration based on it:
```es6
// Adjust loop duration if it's needed
if ( loopDuration > max ) {
	onLoopDurationChange( max );
}
```

It does the same thing you suggested but avoids using the useEffect() hook.

Reply:
Updated